src/Microsoft.AspNetCore.Server.Kestrel/Http/FrameOfT.cs

```diff
@@ -144,6 +135,8 @@ public class Frame<TContext> : Frame
             }
             finally
             {
+                // Error occurred, do not return components to pool
+                ResetComponents(poolingPermitted: false);
```

Comment:
An error didn't necessarily occur here, right? The client could have just closed the connection. Why prevent pooling even if there was an error?